src/coreclr/vm/callingconvention.h

```diff
@@ -620,35 +629,35 @@ class ArgIteratorTemplate : public ARGITERATOR_BASE
 
         pLoc->m_fRequires64BitAlignment = m_fRequires64BitAlignment;
 
-        int cSlots = (GetArgSize() + 3) / 4;
-
+        const int byteArgSize = GetArgSize();
         if (TransitionBlock::IsFloatArgumentRegisterOffset(argOffset))
         {
-            pLoc->m_idxFloatReg = (argOffset - TransitionBlock::GetOffsetOfFloatArgumentRegisters()) / 4;
-            pLoc->m_cFloatReg = cSlots;
+            const int floatRegOfsInBytes = argOffset - TransitionBlock::GetOffsetOfFloatArgumentRegisters();
+            _ASSERTE((floatRegOfsInBytes % FLOAT_REGISTER_SIZE) == 0);
+            pLoc->m_idxFloatReg = (argOffset - TransitionBlock::GetOffsetOfFloatArgumentRegisters()) / FLOAT_REGISTER_SIZE;
```

Comment:
A nit - can you please use the floatRegOfsInBytes constant here too?

Reply:
My bad, thanks for catching